src/libraries/Microsoft.PowerFx.Core/Types/Enums/EnumStoreBuilder.cs

```diff
@@ -284,15 +285,23 @@ internal sealed class EnumStoreBuilder
             };
         #endregion
 
+        // Map from enum name to its type spec. 
         private readonly Dictionary<string, string> _workingEnums = new Dictionary<string, string>();
```

Comment:
>new Dictionary<string, string>(); [](http://example.com/codeflow?start=68&length=33)

Let's use a ConcurrentDictionary<string, string>, that will avoid the lock problem